### PR TITLE
Dw first branch

### PR DESCRIPTION
**Pull request by RocketPuppy**
Hi there. There's a bunch of stuff here, so here's the basic rundown. My goal was to get the Game file in a place where I felt comfortable starting to add a CLI interface to drive it. Mostly, my changes are around encapsulating things. Here's some examples:

* Serialization/Deserialization of XML backed classes is all encapsulated in the `BackingXmlFile` class. You can mostly transparently make new classes serializable in the same way by inheriting from the `Backed` abstract class. I would like to (but haven't yet) move references to the `XML` member to be all internal to their respective class.
* I split up the `Game.Init` method into some component parts. A good size chunk went into calculated properties and initial values for members.  The rest is in the `SpawnFleets` and `SpawnResources` methods.

Hopefully these changes are easy to follow. Let me know if you have any questions. This did build and run fine on my machine, but let me know if there are any issues.

One last thing, I noticed there's mixing of tabs and spaces, which makes the diff a little hard to read. Would you be opposed to me adding a commit on the end of this to standardize on one or the other?

### CODE REVIEW

**Comment by cannonfodder1** [comment]
> One last thing, I noticed there's mixing of tabs and spaces, which makes the diff a little hard to read. Would you be opposed to me adding a commit on the end of this to standardize on one or the other?

Not opposed, I would personally prefer tabs but I can do either.

File: NebulousConquestHelper/src/FileType.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	{
+		public static string ToExtension(this FileType fileType)
+		{
+			if(fileType == FileType.Fleet)
+			{
+				return ".fleet";
+			}
+			if (fileType == FileType.Game)
+			{
+				return ".scenario";
```
Comment (cannonfodder1) [inline NebulousConquestHelper/src/FileType.cs]: I have all of the starting conditions in a .scenario file, but for active campaigns I had the Game class save its data to a .conquest file. They're basically the same, just kept separately labelled for ease of use purposes. That's probably unnecessary, so probably just make everything .conquest files or do some special handling here to support both types if you think that's better.

**Review by cannonfodder1 — COMMENTED** [review]
Overall some very big brain changes, awesome work

File: NebulousConquestHelper/src/FileType.cs (hunk 1 above)
Comment (RocketPuppy) [reply NebulousConquestHelper/src/FileType.cs]: I just changed it all to .conquest. Seemed like the simplest thing for now.

**Comment by RocketPuppy** [comment]
@cannonfodder1 Added the formatting commit

**Pull request merged**